packages/core/tests/location-hash.test.ts

```diff
@@ -0,0 +1,142 @@
+    })
+  })
+
+  describe('url location hash parsing', () => {
+    const dataTransferIdUrlParam = 'id=f773748b-832a-467a-a252-53d711e21ff3'
+    const encryptedPkUrlParam = 'epk=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAxw'
+    const factorsUrlParam = 'f=1,2,3'
+    const walletAddressUrlParam = 'wa=0x999999cf1046e68e36E1aA2E0E07105eDDD1f08E'
+
+    const baseUrlHash = [
+      dataTransferIdUrlParam,
+      encryptedPkUrlParam,
+      factorsUrlParam,
+      walletAddressUrlParam,
+    ].join('&')
+
+    describe('location hash parsing', () => {
+      it('parses with #', () => {
+        const urlWithHashTag = `#${baseUrlHash}`
+
+        const parsedParams = parseUnforgettableLocationHash(urlWithHashTag)
+
+        expect(parsedParams).toEqual({
+          dataTransferId: 'f773748b-832a-467a-a252-53d711e21ff3',
+          encryptionPublicKey: 'MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAxw',
+          factors: [1, 2, 3],
+          walletAddress: '0x999999cf1046e68e36E1aA2E0E07105eDDD1f08E',
+        })
+      })
+
+      it('parses without #', () => {
+        const parsedParams = parseUnforgettableLocationHash(baseUrlHash)
+
+        expect(parsedParams).toEqual({
+          dataTransferId: 'f773748b-832a-467a-a252-53d711e21ff3',
+          encryptionPublicKey: 'MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAxw',
+          factors: [1, 2, 3],
+          walletAddress: '0x999999cf1046e68e36E1aA2E0E07105eDDD1f08E',
+        })
+      })
+    })
+
+    it('throws an error if the data transfer ID is missing', () => {
+      expect(() =>
+        parseUnforgettableLocationHash(
+          [encryptedPkUrlParam, factorsUrlParam, walletAddressUrlParam].join('&'),
+        ),
+      ).toThrow()
+    })
+
+    it('throws an error if the encrypted public key is missing', () => {
+      expect(() =>
+        parseUnforgettableLocationHash(
+          [dataTransferIdUrlParam, factorsUrlParam, walletAddressUrlParam].join('&'),
+        ),
+      ).toThrow()
+    })
+
+    describe('the factors parsing', () => {
+      it('returns empty array if factors not provided', () => {
+        const parsedParams = parseUnforgettableLocationHash(
+          [dataTransferIdUrlParam, encryptedPkUrlParam].join('&'),
+        )
+
+        expect(parsedParams.factors).toEqual([])
+      })
+
+      it('returns empty array if factors key is provided but empty', () => {
+        const parsedParams = parseUnforgettableLocationHash(
+          [dataTransferIdUrlParam, encryptedPkUrlParam, 'f='].join('&'),
+        )
+
+        expect(parsedParams.factors).toEqual([])
+      })
+
+      it('ignores empty items in the enumeration', () => {
+        const parsedParams = parseUnforgettableLocationHash(
+          [dataTransferIdUrlParam, encryptedPkUrlParam, '&f=1,,2,3'].join('&'),
+        )
+
+        expect(parsedParams.factors).toEqual([1, 2, 3])
+      })
+
+      it('throws an error if the factor is not an integer number', () => {
```

Comment:
also check if the factors order is preserved if there are multiple the same factors, e.g. `2,1,3,1,2,1`